sympy/physics/mechanics/_system.py

```diff
@@ -0,0 +1,937 @@
+    Matrix([
+    [bob_mass*l**2*sin(q_pin)**2, -cart_mass*l*cos(q_pin)],
+    [               l*cos(q_pin),                       1]])
+    >>> simplify(system.forcing)
+    Matrix([
+    [-l*(bob_mass*g*sin(q_pin) + bob_mass*l*u_pin**2*sin(2*q_pin)/2 + F*cos(q_pin))],
+    [                                                         l*u_pin**2*sin(q_pin)]])
+
+    """
+
+    def __init__(self, origin=None, frame=None):
+        if origin is None:
+            origin = Point('inertial_origin')
+        elif not isinstance(origin, Point):
+            raise TypeError('Origin must be an instance of Point.')
+        self._origin = origin
+        if frame is None:
+            frame = ReferenceFrame('inertial_frame')
+        elif not isinstance(frame, ReferenceFrame):
+            raise TypeError('Frame must be an instance of ReferenceFrame.')
+        self._frame = frame
+        self._origin.set_vel(self._frame, 0)
+        self._q_ind = Matrix(1, 0, []).T
+        self._q_dep = Matrix(1, 0, []).T
+        self._u_ind = Matrix(1, 0, []).T
+        self._u_dep = Matrix(1, 0, []).T
+        self._kdes = Matrix(1, 0, []).T
+        self._hol_coneqs = Matrix(1, 0, []).T
+        self._nonhol_coneqs = Matrix(1, 0, []).T
+        self._bodies = []
+        self._joints = []
+        self._loads = []
+        self._eom_method = None
+
+    @classmethod
+    def from_newtonian(cls, newtonian):
+        """Constructs the system with respect to a Newtonian body."""
+        if isinstance(newtonian, Particle):
+            raise TypeError('A Particle has no frame so cannot act as '
+                            'the Newtonian.')
+        system = cls(origin=newtonian.masscenter, frame=newtonian.frame)
+        system.add_bodies(newtonian)
+        return system
+
+    @property
+    def origin(self):
+        """Global origin of the system."""
+        return self._origin
+
+    @property
+    def frame(self):
+        """Inertial reference frame of the system."""
+        return self._frame
+
+    @property
+    def x(self):
+        """Unit vector in the x direction of the inertial reference frame."""
+        return self._frame.x
+
+    @property
+    def y(self):
+        """Unit vector in the y direction of the inertial reference frame."""
+        return self._frame.y
+
+    @property
+    def z(self):
+        """Unit vector in the z direction of the inertial reference frame."""
+        return self._frame.z
+
+    @property
+    def bodies(self):
+        """Tuple of all bodies that have been added to the system."""
+        return tuple(self._bodies)
+
+    @bodies.setter
+    @_reset_eom_method
+    def bodies(self, bodies):
+        bodies = System._objects_to_list(bodies)
+        self._check_objects(bodies, [], BodyBase, 'Bodies', 'bodies')
+        self._bodies = bodies
+
+    @property
+    def joints(self):
+        """Tuple of all joints that have been added to the system."""
+        return tuple(self._joints)
```

Comment:
`bodies` and `loads` have setters but `joints` doesn't, why?

Reply:
What are your thoughts on having this for consistency? Personally if `bodies` and `loads` have setters then I'd naturally assume that `joints` would have one too.

Reply:
Addressed in 77ce60400d6e23b2d0d0bad1343c57d7564433a3